### ISSUE TITLE
Add a waiter for Athena query execution to be finished

### ISSUE THREAD

**Issue by alfaproject** [aws/aws-sdk#677]
### Describe the feature

Athena queries are run in the background and there are libraries like Athena Express (which is abandoned) that basically wait until the query is ready to then either download the results or parse them some other way

I think it would be beneficial to have an official waiter for this use case. The concept of waiters is not new to the SDK so it would be just another one, I suppose.

### Use Case

Start an Athena query and wait for it to be completed (not queued and not running) to get the execution result

### Proposed Solution

We have our own simple layer on top of the SDK to abstract some things, and this is how we implemented this suggestion:
```ts
export const startAthenaQueryExecution = (input: StartQueryExecutionCommandInput) =>
  athenaClient().send(new StartQueryExecutionCommand(input));

export const getAthenaQueryExecution = (input: GetQueryExecutionCommandInput) =>
  athenaClient().send(new GetQueryExecutionCommand(input));

export async function executeAthenaQuery(input: StartQueryExecutionCommandInput): Promise<QueryExecution> {
  const { QueryExecutionId: queryExecutionId } = await startAthenaQueryExecution(input);

  let retryIn = 0;
  let queryExecution: QueryExecution;
  do {
    // The query runs asynchronously so wait a bit before checking the result up to 1s per retry
    if (retryIn < 1000) {
      retryIn += 250;
    }
    await setTimeout(retryIn);

    const getQueryExecutionOutput = await getAthenaQueryExecution({ QueryExecutionId: queryExecutionId });
    queryExecution = getQueryExecutionOutput.QueryExecution!;
  } while (
    queryExecution.Status!.State === QueryExecutionState.QUEUED ||
    queryExecution.Status!.State === QueryExecutionState.RUNNING
  );
  return queryExecution;
}
```

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### SDK version used

3.485.0

### Environment details (OS name and version, etc.)

Lambda

**Comment by RanVaknin** [comment]
I have confirmed that the service team has added this to their backlog. Since no further action can be done by the SDK team, I'm going to go ahead and close this. 

Please keep an eye on subsequent SDK release notes for updates on this.

Thanks again for reaching out!
Ran~

**Comment by alfaproject** [comment]
@RanVaknin thank you for your extensive explanation. Is there somewhere where I can follow that progress or it's internal? If it is internal, would it be ok to keep this open so that I get notified later? It's really hard to follow the SDK daily updates otherwise

**Comment by RanVaknin** [comment]
Hi @alfaproject ,

If you have access to AWS support through the AWS Console you can use this ticket ID `P113121613` to request a follow up from an internal support team member. 

Our team cannot own communication for service related issues so once we verify that a customer request has made it to the right owner, and get a confirmation about will / wont implement we close the issue. 

Sorry for the inconvenience.
Ran~
